x/ibc/04-channel/keeper/handshake.go

```diff
@@ -33,38 +35,45 @@ func (k Keeper) ChanOpenInit(
 	connectionHops []string,
 	portID,
 	channelID string,
+	portCap capability.Capability,
 	counterparty types.Counterparty,
 	version string,
-) error {
+) (capability.Capability, error) {
 	// channel identifier and connection hop length checked on msg.ValidateBasic()
 
 	_, found := k.GetChannel(ctx, portID, channelID)
 	if found {
-		return sdkerrors.Wrap(types.ErrChannelExists, channelID)
+		return nil, sdkerrors.Wrap(types.ErrChannelExists, channelID)
 	}
 
 	connectionEnd, found := k.connectionKeeper.GetConnection(ctx, connectionHops[0])
 	if !found {
-		return sdkerrors.Wrap(connection.ErrConnectionNotFound, connectionHops[0])
+		return nil, sdkerrors.Wrap(connection.ErrConnectionNotFound, connectionHops[0])
 	}
 
 	if connectionEnd.GetState() == connectionexported.UNINITIALIZED {
-		return sdkerrors.Wrap(
+		return nil, sdkerrors.Wrap(
 			connection.ErrInvalidConnectionState,
 			"connection state cannot be UNINITIALIZED",
 		)
 	}
 
+	if !k.portKeeper.Authenticate(ctx, portCap, portID) {
```

Comment:
Using portKeeper's Authenticate function to authenticate port capability against portID name

Reply:
Yes, that's correct.